src/ray/object_manager/plasma/stats_collector.cc

```diff
@@ -24,6 +24,13 @@ namespace plasma {
 void ObjectStatsCollector::OnObjectCreated(const LocalObject &obj) {
   const auto kObjectSize = obj.GetObjectInfo().GetObjectSize();
   const auto kSource = obj.GetSource();
+  const auto &kAllocation = obj.GetAllocation();
+
+  if (kAllocation.fallback_allocated) {
+    num_bytes_on_fallback_ += kObjectSize;
+  } else {
+    num_bytes_on_shared_memory_ += kObjectSize;
+  }
```

Comment:
Consider using CounterMap, and registering a callback that records metrics on change.